### DISCUSSION TITLE
Passing CSS props to functional component

### DISCUSSION THREAD

**Discussion by carldegs** [emotion-js/emotion#2631] [Q&A] [answered]
Hi there!

I am trying to pass the css object prop to a functional component, but it's always returning undefined. What's the correct way to do this? Thanks! 

Here's an example. Let's say I have a component `Component`  which expects a `css` prop that will be added to it's div's `css` prop. 
```JSX
const Component = ({ css, children }) => <div css={{ 'background': 'blue', ...css}}>{children}</div>
```

But when I pass a prop to it in the parent component, I am getting undefined instead.
```JSX
const Page = () => {
  return (
    <div>
      <Component css={{ color: 'white' }}>Test</Component>
    </div>
  );
};
```

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Reply by carldegs** [reply]
Sorry about that. I'll take note of that next time.

**Comment by srmagura** [comment, marked as answer]
Hey @carldegs, the css prop is "magic" so it won't get passed down to `Component` like a normal prop. You can review [this section](https://emotion.sh/docs/css-prop#style-precedence) of our documentation.

This is what you want:

```tsx
const Component = ({ className, children }) => <div css={{ 'background': 'blue' }} className={className}>{children}</div>
```

`className` will include the Emotion-generated class from the css prop.

**Reply by carldegs** [reply]
Thanks a lot @srmagura!
